### ISSUE TITLE
Client side certificates with OIDs larger than 28 bits fail to load/authenticate

### ISSUE THREAD

**Issue by antevens** [helm/helm#7343]
helm repo add sde https://helm.example.com/pulp/Default_Organization/Library/custom/sde/sde-helm/ --cert-file user-key-cert.crt --key-file=user-key-cert.key 
Error: looks like "https://helm.example.com/pulp/Default_Organization/Library/custom/sde/sde-helm/" is not a valid chart repository or cannot be reached: can't create TLS config for client: can't load key pair from cert user-key-cert.crt  user-key-cert.key-key-cert.key: asn1: structure error: base 128 integer too large

Probably related to:
https://github.com/golang/go/issues/19933

Output of `helm version`:
version.BuildInfo{Version:"v3.0.2", GitCommit:"19e47ee3283ae98139d98460de796c1be1e3975f", GitTreeState:"clean", GoVersion:"go1.13.5"}

**Comment by thomastaylor312** [comment]
@antevens So it looks like this could be dependent on operating system (particularly if it is 32 bit). The Go issue linked was over 2 years ago so the bug shouldn't be present in the Go version we use. What OS are you using?

**Comment by thomastaylor312** [comment]
For reference, this is the patch I found, which shows 2 possible failure states that would give you the error reported above: https://go-review.googlesource.com/c/go/+/40436/2/src/encoding/asn1/asn1.go

**Comment by antevens** [comment]
Hey Thomas,

Let me know if there is any additional information I can provide or steps to take to help debug this:

astevens@host:~$ uname -a
Linux host 5.3.6-050306-generic #201910111731 SMP Fri Oct 11 17:34:45 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
astevens@host:~$ cat /etc/lsb-release 
DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=19.1
DISTRIB_CODENAME=tessa
DISTRIB_DESCRIPTION="Linux Mint 19.1 Tessa"
astevens@host:~$ snap --version
snap       2.42.5
snapd      2.42.5
series     16
linuxmint  19.1
kernel     5.3.6-050306-generic
astevens@host:~$ 

On the server side I'm distributing the packages using a Foreman (RedHat Satellite) server, I can't provide you with the exact key/cert that I was using for security reasons, working on switching to a test server where I can provide a cert/key to test.

**Comment by antevens** [comment]
I've created a test cert/key which can be downloaded here and used to reproduce the issue:
https://gist.github.com/antevens/a05409165d33bd771e39bc219fe2c6bb/archive/e56cf114dee64a6b367abf08ab1c34a405a62f88.zip

**Comment by thomastaylor312** [comment]
Thanks for the info! I'll take a look tomorrow

**Comment by thomastaylor312** [comment]
So I gave it a try and got the same error. It is coming from the section of `encoding/asn1` I thought it would: https://golang.org/src/encoding/asn1/asn1.go#L309. Basically it is detecting that the number of bit shifts is greater than the number of bits allowed for an `int32`. I am 99% certain that this is likely a Go issue. All we are doing in Helm is convenience wrapping the lower level `tls.LoadX509KeyPair`, so we aren't modifying anything that would cause the issue. To double check, I created this simple go program and tried to load the example certs you gave me:
```go
```
So I am going to close this issue for now. I'd recommend bringing it up in the Go issues
